core/client/fs/src/main/java/alluxio/client/file/options/OutStreamOptions.java

```diff
@@ -69,19 +70,35 @@ public static OutStreamOptions defaults() {
    */
   public OutStreamOptions(CreateFilePOptions options) {
     this();
-    mBlockSizeBytes = options.getBlockSizeBytes();
-    try {
-      mLocationPolicy = (FileWriteLocationPolicy) CommonUtils.createNewClassInstance(
-          Class.forName(options.getFileWriteLocationPolicy()), new Class[] {}, new Object[] {});
-    } catch (Exception e) {
-      throw new RuntimeException(e);
+    if (options.hasBlockSizeBytes()) {
+      mBlockSizeBytes = options.getBlockSizeBytes();
+    }
+    if (options.hasMode()) {
+      mMode = new Mode((short) options.getMode());
+    }
+    if (options.hasReplicationDurable()) {
+      mReplicationDurable = options.getReplicationDurable();
+    }
+    if (options.hasReplicationMin()) {
+      mReplicationMin = options.getReplicationMin();
+    }
+    if (options.hasReplicationMax()) {
+      mReplicationMax = options.getReplicationMax();
+    }
+    if (options.hasWriteTier()) {
+      mWriteTier = options.getWriteTier();
+    }
+    if (options.hasWriteType()) {
```

Comment:
If the options are not set, do these member variables get set correctly?

Reply:
Yes, the OutStreamOptions::_ctor reads from Configuration.